gbpservice/neutron/services/servicechain/drivers/oneconvergence_intercloud_driver.py

```diff
@@ -206,6 +191,24 @@ def _fetch_template_and_params(self, context, sc_instance,
                                     hotplugged_port_list[0]['mac_address']]})
                 stack_template[resources_key]['Firewall'][properties_key][
                     'description'] = str(firewall_desc)
+            else:
+                config_param_values['Subnet'] = stitching_subnet_id
+
+                #REVISIT [MAK] - Router Id is mandatory param for vpnservice
+                #External policy will create a router for the tenant.
+                #Not sure how to get External Policies here.. so just
+                #getting any router of that tenant.. If there are no other
+                #then we will get what external policy has created.
+                filters = {'tenant_id': [provider_ptg['tenant_id']]}
+                routers = self._l3_plugin.get_routers(context._plugin_context, filters)
+
+                config_param_values['RouterId'] = routers[0]['id']
+
+                provider_subnet = self._core_plugin.get_subnet(context._plugin_context, provider_subnet_id)
+
+                desc = 'fip=' + floating_ip + ";" + "tunnel_local_cidr=" + provider_subnet['cidr']
```

Comment:
This code is not with the service manager. Instead of double effort of using "LoadBalancer:DeviceManager" module inside "vpn_device_driver", floating ip was being passed in service desc.
Once the service manager is integrated this code line will be removed.